mixer/pkg/runtime/dispatcher.go

```diff
@@ -314,7 +314,9 @@ func (m *dispatcher) Preprocess(ctx context.Context, requestBag attribute.Bag, r
 		},
 	)
 
-	log.Debugf("Attributes generated from preprocess phase are %v", responseBag.DebugString())
+	if log.DebugEnabled() {
+		log.Debugf("Attributes generated from preprocess phase are %v", responseBag.DebugString())
```

Comment:
As a side question and alternate fix for this problem, why is there a DebugString instead of a String() on responseBag ? that way 
```
log.Debugf("Attributes generated from preprocess phase are %v", responseBag)
```
would just be cheap

or if you need 2 types of "to string" then use %#v and `GoString()` (https://golang.org/pkg/fmt/#GoStringer)

Reply:
I have a different PR that will do just that.